framework/src/mesh/MooseMesh.C

```diff
@@ -2076,6 +2076,12 @@ MooseMesh::buildMeshBaseObject(ParallelType override_type)
   if (!getParam<bool>("allow_renumbering"))
     mesh->allow_renumbering(false);
 
+  // MeshGenerators may call `prepare_for_use` which would delete remote elements before we've had a
+  // chance to add geometric ghosting functors. (You can look at our task list in Moose.C to see
+  // that we execute mesh generators before addition of geometric ghosting). We need to delay remote
+  // element deletion until after we've had a chance to add those functors
+  mesh->allow_remote_element_removal(false);
```

Comment:
@friedmud was concerned about not having the ability to control this. Should we have a parameter for saying, no - really delete for advanced usage?

Reply:
It sounds like @friedmud would prefer to have the default be to allow remote element removal, and I don't necessarily disagree. When I put this up, it was for like a few week stop-gap until we arrived at a more mature design idea. I wouldn't be scarred if we decided not to move forward with this PR although it seems like there are some who would like it if we did, e.g. @YaqiWang @arovinelli.

My preferred design is to always allow remote element deletion but as I've said allow `MeshGenerators` do add ghosting functors.

Reply:
> My preferred design is to always allow remote element deletion but as I've said allow MeshGenerators to add ghosting functors.

Agreed - this is the right fix even if it's not immediately obvious that we need it.

Reply:
So maybe I'll head in that direction with this PR, or should I wait until the 16th?